### PR TITLE
Revise README.md

### PR DESCRIPTION
**Pull request by gayashanbc**
$subject

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,233 @@
...
+4. Fill out the following fields.
+
+       Client ID - <Enter the copied value of `OAuth Client Key` when creating the Service Provider>
+       Client Secret - <Enter the copied value of `OAuth Client Secret` when creating the Service Provider>
+       Authorization Endpoint - https://localhost:9443/oauth2/authorize
+       Token Endpoint - https://localhost:9443/oauth2/token
+       Userinfo Endpoint - https://localhost:9443/oauth2/userinfo
+       Logout Endpoint - https://localhost:9443/oidc/logout
+       Redirect URI - http://localhost:8080/callback/
+       PostLogout Redirect URI - http://localhost:8080/postlogout/
```
Comment (ChamathNS) [inline README.md:65]: Shouldn't the app context be added to these URIs here?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,233 @@
...
+5. Continue the on-screen guidance and complete the installation.
+
+### Running the sample
+Double click on the `Pickup Manager` application available on your Desktop.<br/>
+![pickup manager](https://user-images.githubusercontent.com/15249242/95334396-aa17c580-08cb-11eb-83ee-3b88b8512f68.gif)
+
+## How it works
+This section explains detailed walkthrough on how key aspects handled in the Asgardio .NET OIDC SDK.
```
Comment (ChamathNS) [inline README.md:73]: explains a*

Comment (ChamathNS) [reply README.md:73]: how key aspects are*

File: README.md [hunk 3]
```diff
@@ -1,2 +1,233 @@
...
+
+  * [Trigger authentication](#trigger-authentication)
+  * [Retrieve user attributes](#retrieve-user-attributes)
+  * [Trigger logout](#trigger-logout)
+
+The structure of the sample would be as follows:<br/>
+![structure](https://user-images.githubusercontent.com/15249242/95333060-e34f3600-08c9-11eb-91e3-8ea64446eb77.PNG)
+
+### Trigger authentication
+In the `<APP_HOME>/LoginPage.xaml` page, we have registered a `Click` event named `LoginButton_Click` for the login button to trigger a OIDC authentication:
```
Comment (ChamathNS) [inline README.md:84]: to trigger an*

File: README.md [hunk 4]
```diff
@@ -1,2 +1,233 @@
...
+```xml
+<Button x:Name ="login" Content="LOGIN" HorizontalAlignment="Left" Margin="73,452,0,0" VerticalAlignment="Top" 
+                Width="240" Height="56" FontFamily="Segoe UI Black" FontSize="22" Click="LoginButton_Click" Background="LightSeaGreen" Foreground="Black"/>
```
Comment (ChamathNS) [inline README.md:87]: Without cluttering the Button tag with style elements, can't we use a class? That way it'd be easier for the reader I feel. WDYT?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,233 @@
...
+```
+
+The button click would trigger an authentication request, and redirect the user to the IdP authentication page.
+Upon successful authentication, the user would be redirected to the application homepage.
+
+### Retrieve user attributes
+In the `<APP_HOME>/LoginPage.xaml.cs` file, we have added the following code inside the login button click trigger method to get the user subject value and the user attributes referring the SDK API.
```
Comment (ChamathNS) [inline README.md:94]: Would be more readable if the "login button click" trigger method was differentiated from the rest of the text. Maybe we could use the method name itself like `LoginButton_Click`.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,233 @@
...
+    home.Show();
+    this.Close();
+}
+```
+
+### Trigger logout
+In the `<APP_HOME>/HomePage.xaml` file, we have added the following button to trigger a SLO flow:
+```xml
+<Button x:Name="logoutButton" Width="Auto" Foreground="White" Background="Black" FontWeight="Bold" 
+        FontSize="14" Click="Logout_button_click" Margin="944,10,9.6,650.2">
```
Comment (ChamathNS) [inline README.md:120]: Better if we could use a style class as mentioned before.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,233 @@
...
+        <Image Source="Assets/logout-white.png" Width="20" Height="20"/>
+    </StackPanel>
+</Button>
+```
+
+Clicking on the logout link would trigger the SLO flow.
+
+## Integrating OIDC SDK to your existing .NET application
+This section will guide you on integrating OIDC into your existing .NET application with the Asgardio Dotnet OIDC SDK.
+This allows a .NET application (i.e. Service Provider) to connect with an IDP using OpenID Connect protocol.
```
Comment (ChamathNS) [inline README.md:131]: using *the OpenID

File: README.md [hunk 8]
```diff
@@ -1,2 +1,233 @@
...
+- [Prerequisites](#prerequisites)
+- [Installing the SDK](#installing-the-sdk)
+  * [Using Nuget Package Manager](#using-nuget-package-manager)
+  * [Using the library DLL](#using-the-library-dll)
+- [Login](#login)
+- [Logout](#logout)
+- [Get User Info](#get-user-info)
+
+### Introduction
+These instructions will guide you on integrating OIDC into your .NET application with the Asgardio Dotnet OIDC SDK. This allows an application (i.e. Service Provider) to connect with an IDP using OIDC protocol.
```
Comment (ChamathNS) [inline README.md:143]: This is mentioned in L130-131 as well. Do we need to repeat it here?

File: README.md [hunk 9]
```diff
@@ -1,2 +1,233 @@
...
+3. Include it with the suggested required dependencies for the project/solution. 
+
+Alternatively, you can also run the following command in the Package Manager CLI as shown below. 
+
+``Install-Package io.asgardio.dotnet.oidc.sdk -Version 1.0.0``
+
+#### Using the library DLL
+To build the project from the source, follow the instructions given below.
+
+1. Download the `io.asgardio.dotnet.oidc.sdk.dll` file from [here](link).
```
Comment (ChamathNS) [inline README.md:163]: update link

File: README.md [hunk 10]
```diff
@@ -1,2 +1,233 @@
...
+<configuration>
+    <appSettings>
+        <add key="ClientId" value="fXZNpFBa3aNOCQ6rKU8ldsxT_WAa" />
+        <add key="ClientSecret" value="6x2fQgrT_Ov2R2OMbzfJ5yQjvVEa" />
+        <add key="AuthorizationEndpoint" value="https://localhost:9443/oauth2/authorize" />
+        <add key="TokenEndpoint" value="https://localhost:9443/oauth2/token" />
+        <add key="UserInfoEndpoint" value="https://localhost:9443/oauth2/userinfo" />
+        <add key="LogoutEndpoint" value="https://localhost:9443/oidc/logout" />
+        <add key="RedirectURI" value="http://localhost:8080/callback/" />
+        <add key="PostLogoutRedirectURI" value="http://localhost:8080/postlogout/" />
```
Comment (ChamathNS) [inline README.md:179]: Concern with the app context path as mentioned above

File: README.md [hunk 11]
```diff
@@ -1,2 +1,233 @@
...
+- [Getting Started](#getting-started)
+- [How it works](#how-it-works)
+- [Integrating OIDC SDK to your existing .NET application](#integrating-oidc-sdk-to-your-existing-net-application)
+- [Building from the source](#building-from-the-source)
+- [Contributing](#contributing)
+  * [Reporting issues](#reporting-issues)
+- [License](#license)
+
+## Getting started
+You can experience the capabilities of Asgardio .NET OIDC SDK by following this small guide which contains main sections listed below.
```
Comment (ChamathNS) [inline README.md:20]: TYPO: *as listed below

Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 10 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 9 above)
Comment (gayashanbc) [reply README.md:160]: I will update once release artifacts are pushed

File: README.md (hunk 8 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 7 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 5 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 3 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 2 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 1 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006

File: README.md (hunk 4 above)
Comment (gayashanbc) [reply README.md:87]: Created https://github.com/wso2/product-is/issues/9803 to track this improvement

File: README.md (hunk 6 above)
Comment (gayashanbc) [reply README.md:120]: Created https://github.com/wso2/product-is/issues/9803 to track this improvement

**Review by darshanasbg — APPROVED** [review]

**Pull request merged**